technical-reference/reference-technical-articles/technical-doc-token-converter.md

```diff
@@ -0,0 +1,226 @@
+
+- `tokenAddressIn` and `tokenAddressOut`: address of the token accepted by the converter in the conversion, and the address of the token sent to the user/wallet from the converter in the conversion
+- `incentive`: percentage used to increase the amount finally sent to the user (see below).
+- `conversionAccess` (see more about private conversions below):
+    - `NONE`: Conversion is disabled for the pair
+    - `ALL`: Conversion is enabled for private conversion and users
+    - `ONLY_FOR_CONVERTERS`: Conversion is enabled only for private conversion
+    - `ONLY_FOR_USERS`: Conversion is enabled only for users
+
+For example, in the `XVSVaultConverter` there would be a configuration entry with these values:
+
+- `tokenAdressIn`: XVS token address (the converter accepts XVS)
+- `tokenAddressOut`: BTCB token address (the converter offers BTCB)
+- `incentive`: 0 (initially there won't be any incentive)
+- `conversionAccess`: ALL (everyone can perform conversions of XVS for BTCB in the `XVSVaultConverter`)
+
+## Incentives
+
+To incentivize conversions we allow an increase in the final amount sent out from the converter contract. Example:
+
+- Given:
+    - Conversion rate provided by the oracle: 1 XVS = 5 USDC
+    - Incentive: 10%
+- If the user sends 1 XVS to the contract, they will receive 5.5 USDC (5 USDC base + 0.5 USDC applying the incentive).
+
+The incentive is applied on the base out amount calculated using the conversion rate, calculated with the oracle prices. Incentives will be defined using normal VIP’s, following the Governance mechanism. Each pair of tokens can have a different incentive value. Bigger incentives can be set for converting tokens with lower liquidity, for example.
+
+## Interaction with the token converter contracts
+
+### Get amount out
+
+View function `getAmountOut(uint256 amountInMantissa, address tokenAddressIn, address tokenAddressOut) returns (uint256 amountConvertedMantissa, uint256 amountOutMantissa)`, to request the amount of `tokenAddressOut` tokens that a sender would receive providing `amountInMantissa` tokens of `tokenAddressIn`, where:
+
+- Params:
+    - `amountInMantissa`: the amount of `tokenAddressIn` tokens the sender would provide to perform the conversion. It is defined in terms of the mantissa of `tokenAddressIn`
+    - `tokenAddressIn`: the address of the token provided by the sender to get tokens of `tokenAddressOut`
+    - `tokenAddressOut`: the address of the token to get after the conversion
+- Returned values:
+    - `amountConvertedMantissa`: the amount of `tokenAddressIn` that would be really transferred from the sender to the contract. If there is enough liquidity of `tokenAddressOut` in the contract (use the virtual function `balanceOf(address token)`), this amount is equal to `amountInMantissa`. Otherwise, it will be less than that amount
```

Comment:
```suggestion
    - `amountConvertedMantissa`: the amount of `tokenAddressIn` that would ultimately be transferred from the sender to the contract. This will be the lesser of the converter's liquidity (available with `balanceOf(address token)`) or `amountConvertedMantissa`.
```